security/security-api/built-in-db-identity-store/src/main/java/javaeetutorial/built_in_db_identity_store/PlaintextPasswordHash.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+ *
+ * You may not modify, use, reproduce, or distribute this software except in
+ * compliance with  the terms of the License at:
+ * https://github.com/javaee/tutorial-examples/LICENSE.txt
+ */
+ 
+package javaeetutorial.built_in_db_identity_store;
+import java.util.Map;
+
+import javax.enterprise.context.Dependent;
+import javax.security.enterprise.identitystore.PasswordHash;
+
+/**
+ * Plain text password hash, being defined here just to demonstrate how PasswordHash
+ * can be implemented,should never be used in a production setting or even in legacy environment.
+ */
+@Dependent
+public class PlaintextPasswordHash implements PasswordHash {
```

Comment:
As noted elsewhere, we should not use this `PasswordHash` implementation, or even suggest that plaintext passwords are a possibility. Instead, we should suggest that doing password hashes correctly is hard, and that people should use an existing mechanism, like the built-in `Pbkdf2PasswordHash`, unless they have very specific requirements that can't be met any other way.

That said, if we want to illustrate how to implement a `PasswordHash`, we could show a `DelegatingPasswordHash` instead:
```
/**
 * Password hashing must be done properly to be secure.
 *
 * Consider using the built-in Pbkdf2PasswordHash, or another
 * known-secure implementation, unless you have specific
 * requirements that can't be met by an existing implementation.
 * Make sure you understand the unique requirements of secure
 * password hashing, and how password hashing is different
 * from other uses of hashing.
 */
public class DelegatingPasswordHash implements PasswordHash {

    private Pbkdf2PasswordHash pbkdf2 = new Pbkdf2PasswordHash();

    public DelegatingPasswordHash() {}

    @Override
    public void initialize(Map<String, String> parameters) {
        pbkdf2.initialize(parameters);
    }
    
    @Override
    public String generate(char[] password) {
        return pbkdf2.generate(password);
    }

    @Override
    public boolean verify(char[] password, String hashedPassword) {
        return pbkdf2.verify(password, hashedPassword);
    }

}
```
The above shows how the interface works, without using an anti-pattern for the actual hash, and also acquaints readers with the built-in password hash. You could include comments explaining what the various methods are for, but that would probably just be duplicative of the javadoc.

Reply:
I have modified database example to use `Pbkdf2PasswordHash`. Thanks for the example shown above.  I definitely intend to include the `DelegatingPasswordHash` in one of the examples. Due to paucity of time, I am not using this in db related example.